org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/TestExecutorTest.xtend

```diff
@@ -0,0 +1,104 @@
+package org.testeditor.web.backend.persistence
+
+import ch.qos.logback.classic.Level
+import ch.qos.logback.classic.Logger
+import ch.qos.logback.classic.spi.ILoggingEvent
+import ch.qos.logback.classic.spi.LoggingEvent
+import ch.qos.logback.core.Appender
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.mockito.ArgumentCaptor
+import org.mockito.InjectMocks
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.MockitoAnnotations
+import org.slf4j.LoggerFactory
+import org.testeditor.web.backend.persistence.workspace.WorkspaceProvider
+
+import static org.assertj.core.api.Assertions.*
+import static org.mockito.Mockito.*
+
+class TestExecutorTest {
+
+	@Rule
+	public var temporaryFolders = new TemporaryFolder
+
+	@Mock
+	WorkspaceProvider workspaceProviderMock
+
+	@InjectMocks
+	TestExecutorProvider testExecutorProvider // class under test
+	
+	@Mock
+	var Appender<ILoggingEvent> logAppender
+	var ArgumentCaptor<LoggingEvent> logCaptor
+
+	@Before
+	def void setupMocks() {
+		MockitoAnnotations.initMocks(this)
+		Mockito.when(workspaceProviderMock.workspace).thenReturn(temporaryFolders.root)
+	}
+
+	@Before
+	def void setupLogging() {
+		logCaptor = ArgumentCaptor.forClass(LoggingEvent)
+		val logBackRootLogger = LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME) as Logger
+		logBackRootLogger.addAppender(logAppender)
+	}
+
+	@Test
+	def void testWrongFilename() {
+		// given 
+		temporaryFolders.newFile('actual-file.tl')
+
+		// when
+		val processBuilder = testExecutorProvider.testExecutionBuilder('actual-file.tl')
+
+		// then
+		assertThat(processBuilder).isNotNull
+		verify(logAppender).doAppend(logCaptor.capture)
+		assertThat(logCaptor.value).satisfies [
+			assertThat(formattedMessage).isEqualTo("File 'actual-file.tl' is no test case (does not end on tcl)")
+			assertThat(level).isEqualTo(Level.WARN)
+		]
+	}
+
+	@Test
+	def void testNonExistingFile() {
+
```

Comment:
// given
    val nonExistingFile = 'fantasy-file.tcl'
    …
?
(just for readability)